src/components/jobManager.ts

```diff
@@ -236,7 +236,6 @@ export class JobManager {
 
 				for (const jobHash of otherJobHashes) {
 					this.#uriHashJobHashSetManager.delete(uriHash, jobHash);
-					this.#jobMap.delete(jobHash);
```

Comment:
I would say delete only if there are no other uriHashes that depend on that job.

Reply:
Umm, I see that this whole for loop is removed in `main`.